gitrevise/merge.py

```diff
@@ -199,29 +202,131 @@ def merge_blobs(
         if err.returncode < 0:
             raise
 
+        preimage = err.output
+        recorded_postimage, conflict_id = reuse_recorded_resolution(repo, preimage)
+        if recorded_postimage:
+            return recorded_postimage
+
         # At this point, we know that there are merge conflicts to resolve.
         # Prompt to try and trigger manual resolution.
         print(f"Conflict applying '{labels[2]}'")
         print(f"  Path: '{path}'")
         if input("  Edit conflicted file? (Y/n) ").lower() == "n":
-            raise MergeConflict("user aborted")
+            raise MergeConflict("user aborted")  # pylint: disable=W0707
 
         # Open the editor on the conflicted file. We ensure the relative path
         # matches the path of the original file for a better editor experience.
         conflicts = tmpdir / "conflict" / path.relative_to("/")
         conflicts.parent.mkdir(parents=True, exist_ok=True)
-        conflicts.write_bytes(err.output)
+        conflicts.write_bytes(preimage)
         merged = edit_file(repo, conflicts)
 
         # Print warnings if the merge looks like it may have failed.
-        if merged == err.output:
+        if merged == preimage:
             print("(note) conflicted file is unchanged")
 
-        if b"<<<<<<<" in merged or b"=======" in merged or b">>>>>>>" in merged:
+        if (
+            b"<" * MARKER_SIZE in merged
+            or b"=" * MARKER_SIZE in merged
+            or b">" * MARKER_SIZE in merged
+        ):
             print("(note) conflict markers found in the merged file")
 
         # Was the merge successful?
         if input("  Merge successful? (y/N) ").lower() != "y":
-            raise MergeConflict("user aborted")
+            raise MergeConflict("user aborted")  # pylint: disable=W0707
+
+        record_resolution(repo, conflict_id, preimage, merged)
 
     return Blob(current.repo, merged)
+
+
+def reuse_recorded_resolution(repo, preimage) -> Tuple[Optional[Blob], Optional[str]]:
+    rr_cache = repo.gitdir / "rr-cache"
+    # TODO Should we cache settings like this?
+    if not repo.bool_config("rerere.enabled", default=rr_cache.is_dir()):
+        return None, None
+
+    conflict_id = conflict_id_by_file_contents(preimage)
+    conflict_dir = rr_cache / conflict_id
+    if not conflict_dir.is_dir():
+        return None, conflict_id
+    if not repo.bool_config("rerere.autoUpdate", default=False):
+        return None, conflict_id
+    postimage_path = conflict_dir / "postimage"
+    postimage = postimage_path.read_bytes()
+    recorded_preimage = (conflict_dir / "preimage").read_bytes()
```

Comment:
These should probably be within a `try: except:` block which warns but does not throw, as we don't want to error out if we fail to read previous rerere conflict resolutions.